stdlib/3/compileall.pyi

```diff
@@ -11,25 +11,67 @@ else:
     _Path = Union[str, bytes, os.PathLike]
     _SuccessType = int
 
-# rx can be any object with a 'search' method; once we have Protocols we can change the type
-def compile_dir(
-    dir: _Path,
-    maxlevels: int = ...,
-    ddir: Optional[_Path] = ...,
-    force: bool = ...,
-    rx: Optional[Pattern[Any]] = ...,
-    quiet: int = ...,
-    legacy: bool = ...,
-    optimize: int = ...,
-    workers: int = ...,
-) -> _SuccessType: ...
-def compile_file(
-    fullname: _Path,
-    ddir: Optional[_Path] = ...,
-    force: bool = ...,
-    rx: Optional[Pattern[Any]] = ...,
-    quiet: int = ...,
-    legacy: bool = ...,
-    optimize: int = ...,
-) -> _SuccessType: ...
-def compile_path(skip_curdir: bool = ..., maxlevels: int = ..., force: bool = ..., quiet: int = ..., legacy: bool = ..., optimize: int = ...) -> _SuccessType: ...
+if sys.version_info >= (3, 7):
+    from py_compile import PycInvalidationMode
+    def compile_dir(
+        dir: _Path,
+        maxlevels: int = ...,
+        ddir: Optional[_Path] = ...,
+        force: bool = ...,
+        rx: Optional[Pattern[Any]] = ...,
+        quiet: int = ...,
+        legacy: bool = ...,
+        optimize: int = ...,
+        workers: int = ...,
+        invalidation_mode: PycInvalidationMode = ...,
```

Comment:
This should be Optional because the default is None; same for the other functions.